### PR TITLE
Fix disabled undo button timeout

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

This PR gets rid of the timeouts we've been seeing in the undo e2e tests. The cause is the `RECEIVE_BLOCKS` action sometimes dispatching between two `UPDATE_BLOCK_ATTRIBUTES` actions. The `RECEIVE_BLOCKS` is not fully ignored: it does not create a new history record, but it _does_ set the `lastAction` attribute. This causes `shouldOverwriteState` to return false where it shouldn't, because it compares the current and previous action.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: packages/editor/src/utils/with-history/test/index.js [hunk 1]
```diff
@@ -167,6 +167,62 @@ describe( 'withHistory', () => {
...
+					ignore: action.content,
+				};
+			}
+
+			return state;
+		};
+
+		const reducer = withHistory( {
+			shouldOverwriteState: ( action, previousAction ) =>
+				previousAction && action.type === previousAction.type,
```
Comment (tofumatt) [inline packages/editor/src/utils/with-history/test/index.js]: Pretty nitpicky but I find this a bit tougher to read; implicit returns I find a bit hard to read when they aren't one-liners or at least wrapped in parens on multiline. How about:
```suggestion
			shouldOverwriteState: ( action, previousAction ) => (
				previousAction && action.type === previousAction.type
			),
```

File: packages/editor/src/utils/with-history/test/index.js [hunk 2]
```diff
@@ -167,6 +167,62 @@ describe( 'withHistory', () => {
...
+			ignoreTypes: [ 'IGNORE' ],
+		} )( complexCounter );
+
+		let state;
```
Comment (tofumatt) [inline packages/editor/src/utils/with-history/test/index.js]: Any reason not to do:

```js
let state = reducer( undefined, {} );
state = reducer( state, { type: 'INCREMENT' } );
```

**Review by tofumatt — APPROVED** [review]
Looks good to me, just little suggestions. Awesome work, the times of restarting timed-out tests is over! 🎉

File: packages/editor/src/utils/with-history/test/index.js (hunk 2 above)
Comment (iseulde) [reply packages/editor/src/utils/with-history/test/index.js]: Just copied another test tbh. 😅

Comment (tofumatt) [reply packages/editor/src/utils/with-history/test/index.js]: it-was-like-that-when-i-got-here

**Pull request merged**